File: 프로그래머스/0/120897. 약수 구하기/약수 구하기.java
import java.util.stream.*;

class Solution {
    public int[] solution(int n) {
        
        return IntStream.rangeClosed(1, n)
                        .filter(i -> n % i == 0)
                        .toArray();
    }
}

// 다른 풀이
 List<Integer> answer = new ArrayList<>();
for(int i=1; i<=n; i++){
    if(n % i == 0){
        answer.add(i);
    }
}
return answer.stream().mapToInt(x -> x).toArray();

File: 프로그래머스/0/120897. 약수 구하기/약수 구하기.py
def solution(n):
    
    return [i for i in range(1, n + 1) if n % i == 0]

# 다른 풀이
return list(filter(lambda v: n % v == 0, [i for i in range(1, n//2+1)])) + [n]

# 다른 풀이
for i in range(1, n + 1):
    if n % i == 0:
        answer.append(i)
